File: app/src/main/java/co/aspirasoft/sams/controller/LoginController.java
package co.aspirasoft.sams.controller;

import co.aspirasoft.sams.model.users.accounts.Account;
import co.aspirasoft.sams.model.users.accounts.AccountsDatabase;

public class LoginController {

    private final String VIEW_INCOMPLETE = "Some fields are incomplete.";
    private final String VIEW_MISMATCH   = "Invalid username/password combination.";
    private final String VIEW_CONNECT_ERR= "Check your internet connection.";
    private final String VIEW_SUCCESS    = "Login successful.";

    private Account account;

    public LoginController(String username, String password){
        account = new Account(username, password);
    }

    public String execute() {
        String username = account.getUsername();
        String password = account.getPassword();

        if (exists(username) && exists(password)) {

            AccountsDatabase db = AccountsDatabase.getInstance();

            if (db.contains(account)) {
                return VIEW_SUCCESS;
            } else {
                return VIEW_MISMATCH;
            }

        } else {
            return VIEW_INCOMPLETE;
        }

    }

    /**
     * Checks if the given string exists and is not empty. Whitespaces
     * are treated as empty characters.
     *
     * @param string the string to be checked
     * @return true if string has a non-empty value
     */
    private boolean exists(String string) {
        if (string != null) {

            // List of whitespace characters
            String[] whitespaces = new String[]{" ", "\n", "\t"};

            // Remove white spaces
            for (String space : whitespaces) {
                string = string.replace(space, "");
            }

            // Return true if string not empty
            if (!string.equals("")) {
                return true;
            }
        }

        // Return false if string null or empty
        return false;
    }
}

File: app/src/main/java/co/aspirasoft/sams/data/LoginController.kt
package co.aspirasoft.sams.data

import co.aspirasoft.sams.model.Account

class LoginController(username: String, password: String) {

    private val VIEW_INCOMPLETE = "Some fields are incomplete."
    private val VIEW_MISMATCH = "Invalid username/password combination."
    private val VIEW_CONNECT_ERR = "Check your internet connection."
    private val VIEW_SUCCESS = "Login successful."
    private val account: Account = Account(username, password)

    fun execute(): String {
        val username = account.username
        val password = account.password
        return if (exists(username) && exists(password)) {
            val db = AccountsDatabase
            if (db.contains(account)) {
                VIEW_SUCCESS
            } else {
                VIEW_MISMATCH
            }
        } else {
            VIEW_INCOMPLETE
        }
    }

    /**
     * Checks if the given string exists and is not empty. Whitespaces
     * are treated as empty characters.
     *
     * @param string the string to be checked
     * @return true if string has a non-empty value
     */
    private fun exists(s: String?): Boolean {
        var string: String? = s
        if (string != null) {

            // List of whitespace characters
            val whitespaces = arrayOf(" ", "\n", "\t")

            // Remove white spaces
            for (space in whitespaces) {
                string = string!!.replace(space, "")
            }

            // Return true if string not empty
            return string != ""
        }

        // Return false if string null or empty
        return false
    }

}